### PR TITLE
Update assets controllers to version 16

### PR DESCRIPTION
**Pull request by pedronfigueiredo**
## **Description**

Updates assets controllers on the extension to the latest version.

### CODE REVIEW

**Comment by jiexi** [comment]
there are breaking changes that need to be adopted 

https://github.com/MetaMask/core/blob/main/packages/assets-controllers/CHANGELOG.md

**Comment by adonesky1** [comment]
Yeah I can add a commit to make some of the adaptations on Monday. About to sign off for the week.

**Review by DDDDDanica — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -263,7 +263,7 @@
     "@metamask/message-manager": "^7.3.0",
     "@metamask/metamask-eth-abis": "^3.0.0",
     "@metamask/name-controller": "^3.0.0",
-    "@metamask/network-controller": "^12.2.0",
+    "@metamask/network-controller": "^15.0.0",
```
Comment (jiexi) [inline package.json:266]: v13 has big breaking changes. The changes necessary to adopt v13 are here: https://github.com/MetaMask/metamask-extension/pull/20652

Comment (adonesky1) [reply package.json:266]: I know I said it might but I'm not sure if this is required for assets-controllers version bump?

**Review by legobeat — CHANGES_REQUESTED** [review]
Looks like some changes are necessary after upgrade:
```
app/scripts/controllers/transactions/IncomingTransactionHelper.test.ts:142:48 - error TS2339: Property 'networkId' does not exist on type 'NetworkState'.

142           currentNetworkId: NETWORK_STATE_MOCK.networkId,
                                                   ~~~~~~~~~

app/scripts/controllers/transactions/IncomingTransactionHelper.ts:280:36 - error TS2339: Property 'networkId' does not exist on type 'NetworkState'.

280     return this.#getNetworkState().networkId as string;
                                       ~~~~~~~~~
```
https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/60144/workflows/dfd963f1-31ff-45be-8fdb-39e12a92b3f5/jobs/1867947

Also needs rebase on `develop`.

**Comment by pedronfigueiredo** [comment]
@metamaskbot update-policies

**Comment by pedronfigueiredo** [comment]
@metamaskbot update-policies

**Review by legobeat — COMMENTED** [review]
Unit test passing

**Review by legobeat — COMMENTED** [review]
Lint passing :+1:

**Review by DDDDDanica — APPROVED** [review]

**Review by legobeat — APPROVED** [review]

**Pull request merged**